File: Clockio/app/src/main/java/com/natasha/clockio/ui/ProfileViewModel.java
package com.natasha.clockio.ui;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.natasha.clockio.data.repository.UserRepository;
import com.natasha.clockio.model.User;

public class ProfileViewModel extends ViewModel {
    private static final String TAG = ProfileViewModel.class.getSimpleName();

    private MutableLiveData<User> userResult = new MutableLiveData<>();
//    private UserRepository userRepository;
//
//    public ProfileViewModel(UserRepository userRepository) {
//        this.userRepository = userRepository;
//    }

    public LiveData<User> getUserResult() {
        return userResult;
    }

    public void getProfile(String accessToken){
        Log.d(TAG, "Profile View Model get Profile");
        userResult = new UserRepository().getProfile(accessToken);
    }
}

File: Clockio/app/src/main/java/com/natasha/clockio/home/viewmodel/ProfileViewModel.kt
package com.natasha.clockio.home.viewmodel

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Transformations
import androidx.lifecycle.ViewModel
import com.natasha.clockio.base.model.BaseResponse
import com.natasha.clockio.home.entity.Employee
import com.natasha.clockio.home.repository.ProfileRepository
import javax.inject.Inject

class ProfileViewModel @Inject constructor(private val profileRepository: ProfileRepository): ViewModel() {
  private val TAG: String = ProfileViewModel::class.java.simpleName

  private val _id = MutableLiveData<String>()
  val id: LiveData<String>
  get() = _id

  val employee: LiveData<BaseResponse<Employee>> = Transformations
      .switchMap(_id) {id ->
        Log.d(TAG, "profileViewModule id $id")
        profileRepository.getEmployeeById(id)
      }

  fun setId(id: String?) {
    if (_id.value != id) {
      _id.value = id
    }
    Log.d(TAG, "profileViewModule _id ${_id.value}")
  }
}
